lib/json_schemer/schema.rb

```diff
@@ -328,7 +327,7 @@ def defs_keyword
     end
 
     def resources
-      @resources ||= { :lexical => {}, :dynamic => {} }
```

Comment:
I think instead of wrapping both of these in a single object, we should pull them into separate objects. Something like:

```ruby
def resources
  @resources ||= { :lexical => Resources.new, :dynamic => Resources.new }
end
```

Or even separate `lexical_resources` and `dynamic_resources` methods.

And then on `Resources` you just have to define the  `Hash` methods that we're using (`#[]`, `#[]=`, `#fetch`, and `#key?`).